internal/storage/bucket_handle.go

```diff
@@ -41,9 +41,15 @@ func (bh *bucketHandle) NewReader(
 	ctx context.Context,
 	req *gcs.ReadObjectRequest) (rc io.ReadCloser, err error) {
 	// Initialising the starting offset and the length to be read by the reader.
-	start := int64((*req.Range).Start)
-	end := int64((*req.Range).Limit)
-	length := int64(end - start)
+	start := int64(0)
+	length := int64(-1)
+	// If Range equals to nil, it has a default call which has start and length as above values otherwise it will take mentioned range.
```

Comment:
rephrase it to  Following the semantics of NewReader method. Passing start, length as 0,-1 reads the entire file.